WMIAdventure/backend/WMIAdventure_backend/IngameUsers/views.py

```diff
@@ -107,25 +135,10 @@ def get_queryset(self):
         user_profile = user.userprofile
         return user_profile.user_decks.all()
 
-    def _give_all_not_owned_cards_to_user(self, user_profile, request_data):
-        """
-        Gives all cards to user if he doesn't own them.
-
-        TODO: REMOVE THIS IN THE FUTURE. For now all users should have all cards, this method will be removed when
-         there will be some way of gaining cards from battle or there will be some other way.
-        """
-
-        # If user is not owner of all cards then give him all cards
-        if user_profile.user_cards.all().count() != Card.objects.all().count():
-            user_cards_ids = user_profile.user_cards.values_list('card_id', flat=True)
-            not_owned_cards = Card.objects.filter(~Q(pk__in=user_cards_ids))
-            new_user_cards = [UserCard(user_profile=user_profile, card=card) for card in not_owned_cards]
-            UserCard.objects.bulk_create(new_user_cards)
-
     def update(self, request, *args, **kwargs):
         deck_to_update = self.get_object()
-        self._give_all_not_owned_cards_to_user(request.user.userprofile,
-                                               request.data)  # TODO: Remove this when gaining cards is implemented
+        # TODO: Remove this when gaining cards is implemented
+        _give_all_not_owned_cards_to_user(request.user.userprofile)

```

Comment:
To powinno się wywoływać przy rejestracji i przy akceptowaniu nowej karty do systemu

Reply:
dodane